include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -4438,6 +4439,73 @@
    */
   ORT_API2_STATUS(SetUserLoggingFunction, _Inout_ OrtSessionOptions* options,
                   _In_ OrtLoggingFunction user_logging_function, _In_opt_ void* user_logging_param);
+
+  /**
+   * Get number of input from OrtShapeInferContext
+   *
+   * \param[in] context
+   * \param[out] out The number of inputs
+   *
+   * \since Version 1.17.
+   */
+  ORT_API2_STATUS(ShapeInferContext_GetInputCount, _In_ const OrtShapeInferContext* context, _Out_ size_t* out);
+
+  /**
+   * Get type and shape info of an input
+   *
+   * \param[in] context
+   * \param[in] index The index of the input
+   * \param[out] info Type shape info of the input
+   *
+   * \since Version 1.17.
+   */
+  ORT_API2_STATUS(ShapeInferContext_GetInputTypeShape, _In_ const OrtShapeInferContext* context, _In_ size_t index, _Outptr_ OrtTensorTypeAndShapeInfo** info);
+
+  /**
+   * Get attribute from OrtShapeInferContext
+   *
+   * \param[in] context
+   * \param[in] attr_name
+   * \param[out] attr
+   *
+   * \since Version 1.17.
+   */
+  ORT_API2_STATUS(ShapeInferContext_GetAttribute, _In_ const OrtShapeInferContext* context, _In_ const char* attr_name, _Outptr_ const OrtOpAttr** attr);
```

Comment:
>ShapeInferContext_GetAttribute [](http://example.com/codeflow?start=18&length=30)

Here we collapse the attribute from the node into ShapeInferenceContext. Maybe should doc that the ShapeInferenceContext is per-node?

Reply:
Done.